File: java_tests/src/main/java/org/infinispan/client/hotrod/configuration/ServerConfiguration.java
package org.infinispan.client.hotrod.configuration;

/**
 * ServerConfiguration.
 *
 * @author Tristan Tarrant
 * @since 5.3
 */
public class ServerConfiguration {
   private cli.Infinispan.HotRod.Config.ServerConfiguration jniServerConfiguration;

   ServerConfiguration(cli.Infinispan.HotRod.Config.ServerConfiguration jniServerConfiguration) {
      this.jniServerConfiguration = jniServerConfiguration;
   }

   public String host() {
      return this.jniServerConfiguration.Host();
   }

   public int port() {
      return this.jniServerConfiguration.Port();
   }

}

File: src/Infinispan.HotRod/Config/ServerConfiguration.cs
using System;

namespace Infinispan.HotRod.Config
{
    /// <summary>
    ///   Used to hold the server specific configurations.
    /// </summary>
    public class ServerConfiguration
    {
        private Infinispan.HotRod.SWIG.ServerConfiguration config;

        internal ServerConfiguration(Infinispan.HotRod.SWIG.ServerConfiguration config)
        {
            this.config = config;
        }

        internal Infinispan.HotRod.SWIG.ServerConfiguration Config()
        {
            return config;
        }

        /// <summary>
        ///   Retrieves the server host.
        /// </summary>
        public String Host()
        {
            return config.getHost();
        }

        /// <summary>
        ///   Retrieves the server port.
        /// </summary>
        public int Port()
        {
            return config.getPort();
        }
    }
}